test/kleros/kleros-liquid-extra-views.js

```diff
@@ -0,0 +1,262 @@
+                depth - 1,
+                K * ID + i + 1,
+                newMinStake,
+                subcourtMap
+              ).subcourtTree
+          )
+        : undefined,
+    hiddenVotes: ID % 2 === 0,
+    jurorFee: randomInt(100),
+    jurorsForJump: randomInt(15, 3),
+    minStake: newMinStake,
+    sortitionSumTreeK: randomInt(2, 5),
+    timesPerPeriod: [...new Array(4)].map(_ => randomInt(5))
+  }
+  if (ID === 0) subcourtTree.parent = 0
+  else {
+    subcourtTree.parent = Math.floor((ID - 1) / K)
+    subcourtMap[subcourtTree.ID] = {
+      ...subcourtTree,
+      children:
+        subcourtTree.children && subcourtTree.children.map(child => child.ID)
+    }
+  }
+  return { subcourtMap, subcourtTree }
+}
+const asyncForEach = async (method, iterable) => {
+  const array = Array.isArray(iterable) ? iterable : Object.values(iterable)
+  for (const item of array) await method(item)
+}
+
+contract('xKlerosLiquid', accounts => {
+  let pinakion
+  let RNG
+  let governor
+  let minStakingTime
+  let maxDrawingTime
+  let subcourtTree
+  let subcourtMap
+  let klerosLiquid
+  let extraViews
+  beforeEach(async () => {
+    governor = accounts[0]
+    // Deploy contracts and generate subcourts
+    pinakion = await Pinakion.new(
+      0x0, // _tokenFactory
+      0x0, // _parentToken
+      0, // _parentSnapShotBlock
+      'Pinakion', // _tokenName
+      18, // _decimalUnits
+      'PNK', // _tokenSymbol
+      true // _transfersEnabled
+    )
+    await pinakion.generateTokens(governor, -1)
+
+    const randomNumber = 10
+    RNG = await ConstantNG.new(randomNumber)
+    minStakingTime = 1
+    maxDrawingTime = 1
+    const {
+      subcourtMap: _subcourtMap,
+      subcourtTree: _subcourtTree
+    } = generateSubcourts(randomInt(4, 2), 3)
+    subcourtTree = _subcourtTree
+    subcourtMap = _subcourtMap
+
+    klerosLiquid = await KlerosLiquid.new(
+      governor,
+      pinakion.address,
+      RNG.address,
+      minStakingTime,
+      maxDrawingTime,
+      subcourtTree.hiddenVotes,
+      subcourtTree.minStake,
+      subcourtTree.alpha,
+      subcourtTree.jurorFee,
+      subcourtTree.jurorsForJump,
+      subcourtTree.timesPerPeriod,
+      subcourtTree.sortitionSumTreeK
+    )
+
+    await pinakion.changeController(klerosLiquid.address)
+    extraViews = await KlerosLiquidExtraViews.new(klerosLiquid.address)
+  })
+
+  it('Should query delayed stakes correctly.', async () => {
+    await klerosLiquid.createSubcourt(
+      subcourtTree.parent,
+      subcourtTree.hiddenVotes,
+      subcourtTree.minStake + 1,
+      subcourtTree.alpha,
+      subcourtTree.jurorFee,
+      subcourtTree.jurorsForJump,
+      subcourtTree.timesPerPeriod,
+      subcourtTree.sortitionSumTreeK
+    )
+
+    const extraData = `0x${(0).toString(16).padStart(64, '0')}${(1)
+      .toString(16)
+      .padStart(64, '0')}`
+    await klerosLiquid.createDispute(2, extraData, {
+      value: await klerosLiquid.arbitrationCost(extraData)
+    })
```

Comment:
Do we really need to create dispute to pass the phase?

Reply:
Yes, because of this https://github.com/kleros/kleros/blob/dae4e63f6e386421dd8ead22411d3be8ba4500ad/contracts/kleros/KlerosLiquid.sol#L393.